src/pudl/transform/ferc1.py

```diff
@@ -717,63 +720,75 @@ def drop_null_data_rows(self, df: pd.DataFrame) -> pd.DataFrame:
-class PlantsSteamFerc1:
+class PlantsSteamFerc1TableTransformer(Ferc1AbstractTableTransformer):
     """Transformer class for the plants_steam_ferc1 table."""
 
-    def execute(
+    table_id: Ferc1TableId = Ferc1TableId.PLANTS_STEAM_FERC1
+
+    def transform(
         self,
         raw_dbf: pd.DataFrame,
         raw_xbrl_instant: pd.DataFrame,
         raw_xbrl_duration: pd.DataFrame,
         transformed_fuel: pd.DataFrame,
     ):
         """Perform table transformations for the plants_steam_ferc1 table."""
-        plants_steam_combo = (
-            self.pre_concat_clean_and_concat_dbf_xbrl(
-                raw_dbf, raw_xbrl_instant, raw_xbrl_duration
-            )
-            .pipe(self.simplify_strings_for_table)
-            .pipe(self.clean_strings_for_table)
-            .pipe(self.oob_to_nan_for_table)
-            .pipe(self.convert_units_in_table)
-            .pipe(self.convert_table_dtypes)
-            .pipe(plants_steam_assign_plant_ids, ferc1_fuel_df=transformed_fuel)
-        )
-        plants_steam_validate_ids(plants_steam_combo)
-        return plants_steam_combo
-
-    def process_dbf(self, raw_dbf):
-        """Modifications of the dbf plants_steam_ferc1 table before concat w/ xbrl."""
-        plants_steam_dbf = self.rename_columns(
-            raw_table=raw_dbf,
-            source="dbf",
-        ).pipe(self.assign_record_id, source_ferc1=Ferc1Source.DBF)
-        return plants_steam_dbf
-
-    def process_xbrl(self, raw_xbrl_instant, raw_xbrl_duration):
-        """Modifications of the xbrl plants_steam_ferc1 table before concat w/ xbrl."""
-        plants_steam_xbrl = (
-            self.merge_instant_and_duration_tables(
-                duration=raw_xbrl_duration,
-                instant=raw_xbrl_instant,
+        self.plants_steam = (
+            self.concat_dbf_xbrl(
+                raw_dbf=raw_dbf,
+                raw_xbrl_instant=raw_xbrl_instant,
+                raw_xbrl_duration=raw_xbrl_duration,
             )
+            .pipe(self.normalize_strings_multicol, params=self.params.normalize_strings)
+            .pipe(self.nullify_outliers_multicol, params=self.params.nullify_outliers)
             .pipe(
-                self.rename_columns,
-                source="xbrl",
+                self.categorize_strings_multicol, params=self.params.categorize_strings
             )
-            .pipe(self.add_report_year_column)
-            .pipe(
-                self.assign_record_id,
-                source_ferc1=Ferc1Source.XBRL,
+            .pipe(self.convert_units_multicol, params=self.params.convert_units)
+            .pipe(self.drop_null_data_rows)
+        )
+        self.plants_steam = plants_steam_assign_plant_ids(
+            self.plants_steam,
+            ferc1_fuel_df=transformed_fuel,
+            fuel_categories=list(
+                FuelFerc1TableTransformer()
+                .params.categorize_strings["fuel_type_code_pudl"]
+                .categories.keys()
```

Comment:
this is what is needed inside of the `PlantsSteamFerc1TableTransformer` to remove the direct linking of param dictionaries within the plant id assignment. The fuel table class is needed here because the categories needed are the fuel table's categories.